data/tutorials/language/1ms_00_modules.md

```diff
@@ -1,545 +1,436 @@
+
+Abstract and read-only types can be either variants, as shown in this section, records, or aliases. It is possible to access a read-only record field's value, but creating such a record requires using a provided function.
 
 ## Submodules
 
 ### Submodule Implementation
 
-We saw that one `example.ml` file results automatically in one module
-implementation named `Example`. Its module signature is automatically derived
-and is the broadest possible, or can be restricted by writing an `example.mli`
-file.
-
-That said, a given module can also be defined explicitly from within a file.
-That makes it a submodule of the current module. Let's consider this
-`example.ml` file:
+A module can be defined inside another module. That makes it a _submodule_.
+Let's consider the files `florence.ml` and `glasgow.ml`
 
+**`florence.ml`**
 ```ocaml
 module Hello = struct
-  let message = "Hello"
-  let hello () = print_endline message
+  let message = "Hello from Florence"
+  let print () = print_endline message
 end
 
-let goodbye () = print_endline "Goodbye"
-
-let hello_goodbye () =
-  Hello.hello ();
-  goodbye ()
+let print_goodbye () = print_endline "Goodbye"
 ```
 
-From another file, it is clear that we now have two levels of modules. We can
-write:
-
-<!-- $MDX skip -->
+**`glasgow.ml`**
 ```ocaml
 let () =
-  Example.Hello.hello ();
-  Example.goodbye ()
+  Florence.Hello.print ();
+  Florence.print_goodbye ()
 ```
 
-### Submodule Interface
+Definitions from a submodule are accessed by chaining module names, here
+`Florence.Hello.print`.
 
-We can also restrict the interface of a given submodule. It is called a module
-type. Let's do it in our `example.ml` file:
+### Submodule With Signatures
 
+To define an interface to a submodule, we can provide a _module signature_. This
+is done in this second version of the `florence.ml` file:
 ```ocaml
 module Hello : sig
- val hello : unit -> unit
-end
-=
-struct
+ val print : unit -> unit
+end = struct
   let message = "Hello"
-  let hello () = print_endline message
+  let print () = print_endline message
 end
 
-(* At this point, Hello.message is not accessible anymore. *)
-
-let goodbye () = print_endline "Goodbye"
-
-let hello_goodbye () =
-  Hello.hello ();
-  goodbye ()
+let print_goodbye () = print_endline "Goodbye"
 ```
 
-The definition of the `Hello` module above is the equivalent of a
-`hello.mli`/`hello.ml` pair of files. Writing all of that in one block of code
-is not elegant so, in general, we prefer to define the module signature
-separately:
+The first version made `Florence.Hello.message` public. In this version it can't be accessed from `glasgow.ml`.
 
-<!-- $MDX skip -->
+### Module Signatures are Types
+
+The role played by module signatures to implementations is akin to the role played by types to values. Here is a third possible way to write file `florence.ml`:
```

Comment:
```suggestion
The module signatures' role to implementations is akin to the role played by types to values. Here is a third possible way to write file `florence.ml`:
```
Is this still correct?

Reply:
Yes it is correct. Can we use possessive for the second part of the sentence, after “akin”, the type & value?